src/components/FeatureViews/AdminPortal/PanelManager.js

```diff
@@ -148,15 +102,65 @@ function PanelManager(props) {
                     fields[field][key] = defaults[key];
                 }
             })
-            console.log(fields)
-            return fields
         })
-    }
+        return fields
+    };
+
+    useEffect(() => {
+        if(records.length !== recordList.length) {
+            let parsedRecordList = records.map((record) => ({
+                ...record,
+                editing: false,
+            }));
+            setRecordList(parsedRecordList);
+        }
+    }, [records]);
     const fieldsWithDefaults = addDefaults(props.fields);
+
+    useEffect(() => {
+        const constructDirectoryObject = (fieldInfo, recordArray) => {
+            recordArray.forEach((record) => {
+                Object.keys(record).forEach((key) => {
+                    if (key !== "id") {
+                        if(fieldInfo[key]["values"]){
+                            fieldInfo[key]["values"].push(record[key])
+                        } else {
+                        fieldInfo[key]["values"] = [record[key]]
+                        }
+                    }
+                })
+            })
+            return fieldInfo
+        }
+
+        setDirectoryData(constructDirectoryObject(fieldsWithDefaults, records));
+    }, [records, fieldsWithDefaults])
+
+    const getRecord = (index) => Object.values(directoryData).map((field) => {
+        console.log("field")
+        console.log(field)
+        return field["values"][index]
+    })
```

Comment:
I'm confused... this function returns a list of records for all fields at a given index? 

Are you trying to get a record from a specific index?